src/renderers/dom/client/wrappers/ReactDOMOption.js

```diff
@@ -32,8 +32,16 @@ var ReactDOMOption = {
 
     // Look up whether this option is 'selected'
     var selectValue = null;
-    if (nativeParent != null && nativeParent._tag === 'select') {
-      selectValue = ReactDOMSelect.getSelectValueContext(nativeParent);
+    if (nativeParent != null) {
+      var selectParent = nativeParent;
+
+      if (selectParent._tag === 'optgroup') {
+        selectParent = selectParent._nativeParent;
+      }
+
+      if (selectParent._tag === 'select') {
```

Comment:
I think we need a null check here, because selectParent could be null at this point, right?